### PR TITLE
Updating docs for 22.04

### PR DESCRIPTION
**Pull request by cjnolet**
Finalizing some more bits of the docs. This has also included cleaning up several header files to make the docs a little more clean.

### CODE REVIEW

**Comment by ajschmidt8** [comment]
Seems like there are a lot more than doc updates in this PR. @cjnolet, can you merge the latest changes to see if that removes some of the extraneous stuff in your PR?

**Comment by cjnolet** [comment]
> Seems like there are a lot more than doc updates in this PR. @cjnolet, can you merge the latest changes to see if that removes some of the extraneous stuff in your PR?

@ajschmidt8, since RAFT is primarily a C++ library, the doc updates in this PR require several updates to the header files as well.

**Comment by ajschmidt8** [comment]
> That said, this PR also includes the changes in https://github.com/rapidsai/raft/pull/540 (which will be merged before this).

Yup, that's likely the culprit here. I'm seeing changes to `build.sh`, `ci/cpu/build.sh` and more in this PR currently. I'll wait until the other PR is merged before reviewing this.

**Comment by cjnolet** [comment]
rerun tests

**Comment by cjnolet** [comment]
rerun tests

**Review by dantegd — APPROVED** [review]
Looks great!

File: docs/source/cpp_api/clustering.rst
Comment (cjnolet) [reply docs/source/cpp_api/clustering.rst:9]: Yep, I think that's a good plan indeed. For 22.04 I think we should keep it the way it is just to avoid any potential breaking changes right before burndown.

**Comment by cjnolet** [comment]
rerun tests

**Comment by cjnolet** [comment]
@gpucibot merge

**Pull request merged**
